orc8r/tools/packer/magma-dev-virtualbox.json

```diff
@@ -1,115 +1,127 @@
+  "builders": [
+    {
+      "boot_command": [
+        "<esc><wait>",
+        "<esc><wait>",
+        "<enter><wait>",
+        "/install/vmlinuz<wait>",
+        " auto<wait>",
+        " console-setup/ask_detect=false<wait>",
+        " console-setup/layoutcode=us<wait>",
+        " console-setup/modelcode=pc105<wait>",
+        " debconf/frontend=noninteractive<wait>",
+        " debian-installer=en_US.UTF-8<wait>",
+        " fb=false<wait>",
+        " initrd=/install/initrd.gz<wait>",
+        " kbd-chooser/method=us<wait>",
+        " keyboard-configuration/layout=USA<wait>",
+        " keyboard-configuration/variant=USA<wait>",
+        " locale=en_US.UTF-8<wait>",
+        " netcfg/get_domain=magma.com<wait>",
+        " netcfg/get_hostname={{ .Name }}<wait>",
+        " grub-installer/bootdev=/dev/sda<wait>",
+        " noapic<wait>",
+        " preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/preseed.cfg<wait>",
+        " -- <wait>",
+        "<enter><wait>"
+      ],
+      "boot_wait": "5s",
+      "guest_additions_mode": "upload",
+      "guest_os_type": "Ubuntu_64",
+      "headless": true,
+      "http_directory": "http",
+      "iso_checksum": "sha256:f11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2",
+      "iso_url": "https://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04/release/ubuntu-20.04.1-legacy-server-amd64.iso",
+      "memory": 8192,
+      "cpus": 4,
+      "hard_drive_interface": "scsi",
+      "name": "magma",
+      "shutdown_command": "echo 'vagrant'|sudo -S shutdown -P now",
+      "ssh_handshake_attempts": "20",
+      "ssh_password": "vagrant",
+      "ssh_timeout": "64206s",
+      "ssh_username": "vagrant",
+      "type": "virtualbox-iso",
+      "vboxmanage": [
+        [
+          "modifyvm",
+          "{{.Name}}",
+          "--memory",
+          "8192"
+        ],
+        [
+          "modifyvm",
+          "{{.Name}}",
+          "--cpus",
+          "4"
+        ]
+      ],
+      "virtualbox_version_file": ".vbox_version",
+      "vm_name": "magma-dev"
+    }
+  ],
+  "post-processors": [
+    [
+      {
+        "output": "builds/magma_dev_focal_{{.Provider}}.box",
+        "type": "vagrant"
+      }
     ]
-  }],
+  ],
   "provisioners": [
-{
-      "type": "shell",
-      "script": "scripts/provision-stretch.sh",
-      "execute_command": "echo 'vagrant' | sudo -S env {{.Vars}} {{.Path}}"
-    },{
-      "type": "shell",
-      "inline": ["sudo reboot"],
-      "expect_disconnect": true
-    },{
-      "type": "shell",
-      "script": "scripts/guest_additions.sh",
+    {
       "execute_command": "echo 'vagrant' | sudo -S env {{.Vars}} {{.Path}}",
-      "pause_before": "10s"
-    }, {
-      "type": "shell",
-      "script": "scripts/vagrant_key.sh",
-      "execute_command": "echo 'vagrant' | sudo -S env {{.Vars}} {{.Path}}"
+      "scripts": [
+        "scripts/ubuntu_setup.sh",
+        "scripts/vagrant_key.sh"
+      ],
+      "type": "shell"
```

Comment:
nitpick: I find the old ordering that was based on semantics easier to understand than the new alphabetical ordering.

For example a provisioning step is easier to read for me if `type` comes first, followed by the most important argument to that type, like `inline` or `script`. Same for other items in the JSON hierarchy like the builders where I would also prefer having the type at the top, or having the post-processors after the provisioners which sort of reflects the order in which things are done.

Sticking to the old ordering has the additional advantage of making this diff much nicer to read ;)

Reply:
Adapted. While we only renamed the `.json` here, I agree that a semantic ordering improves readability and leads to an easier understanding.

Reply:
Thanks! And yes, I actually read the diff wrong because of the rename ;)